src/astra_rl/algorithms/dpo.py

```diff
@@ -0,0 +1,136 @@
+from dataclasses import dataclass
+from typing import Generic, Sequence, List
+
+from astra_rl.core.algorithm import Algorithm
+from astra_rl.core.problem import Problem
+from astra_rl.core.common import StateT, ActionT
+from astra_rl.core.environment import Graph
+
+import torch
+import torch.nn.functional as F
+
+
+@dataclass
+class DPOStep(Generic[StateT, ActionT]):
+    prefix: StateT
+
+    suffix_pos: ActionT
+    suffix_neg: ActionT
+
+
+@dataclass
+class DPOBatch(Generic[StateT, ActionT]):
```

Comment:
Would be useful to do a scrub of the PR for everything that's a definition to add docstrings for the class and method definitions to capture the context and information from the implementation and reasoning while it's all still fresh.